### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

**Comment by gyandeeps** [comment]
@maenu Can you please set up the travis CI build for this repo. As it will build all the PR and the branch master whenever there is a push on the branch.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+indent_style = tab
+indent_size = 4
+trim_trailing_whitespace = true
+end_of_line = crlf
```
Comment (maenu) [inline .editorconfig]: I would like LF here.

Comment (gyandeeps) [inline .editorconfig]: will make that change.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+	"extends": "eslint-config-gyandeep",
```
Comment (maenu) [inline .eslintrc]: Can you give me a link?
How is this resolved?

Comment (gyandeeps) [inline .eslintrc]: New eslint actually resolves it inside its processing.
Repo link: https://github.com/gyandeeps/eslint-config-gyandeep
More about eslint shareable config: http://eslint.org/docs/developer-guide/shareable-configs

Comment (maenu) [inline .eslintrc]: Ok, maybe we should create our own, or extend a default one (if there are any).

Comment (gyandeeps) [inline .eslintrc]: This is just to get started for now. I will make those changes later.

**Comment by maenu** [comment]
Enabled Travis.

I was a bit afraid when I saw that there were so many changed files.
Now I see that all changes are just making the code style more consistent.
I like that ;).

**Comment by gyandeeps** [comment]
Awesome. Thanks for the quick response.
I think this PR is ready to merge, any thoughts?

**Comment by maenu** [comment]
Looks fine for me, but I got two failing tests:

```
$ grunt test
Running "nodeunit:template" (nodeunit) task
Testing template.js.......FF.............
>> template - instrumentation - shouldInstrumentSource
>> Message: should be instrumented
>> Error: 'var __cov_IWp9qM1GTBo1x3ZChf8cEQ = (Function(\'return this\'))();' == 'var __cov_K0_Q1LAyR$FdJWor9Wn_4A = (Function(\'return this\'))();'
>> at Object.exports.template.instrumentation.shouldInstrumentSource (src/test/js/template.js:193:9)
>> at Object.exports.template.setUp (src/test/js/template.js:59:3)

>> template - instrumentation - shouldGenerateUris
>> Message: should have generated source URI
>> Error: '.grunt/temp/C:/some/file.js' == '.grunt/temp/C/some/file.js'
>> at Object.exports.template.instrumentation.shouldGenerateUris (src/test/js/template.js:217:9)
>> at Object.exports.template.setUp (src/test/js/template.js:59:3)

Warning: 2/36 assertions failed (954ms) Use --force to continue.

Aborted due to warnings.
```

I guess the first one isn't a very good assertion anyway, should probably use RegEx.
But the second one looks suspicious, but was broken before.
I have no idea what happened there, but it's not this PR that breaks it, so it's fine.
I'll merge.

**Comment by gyandeeps** [comment]
Hold let me make one more change for travis build

**Comment by maenu** [comment]
That was close, I already had the pointer on the button ;).

**Comment by gyandeeps** [comment]
I am going keep making changes on this PR to make sure travis build is working.
Once I am done I will merge.

**Comment by maenu** [comment]
:+1:

**Pull request merged**